### PR TITLE
Add missing invalidate calls to CachingHiveMetastore + Fix REVOKE in FileHiveMetastore

### PR DESCRIPTION
**Pull request by iirekm**

### CODE REVIEW

File: presto-hive-hadoop2/src/test/java/io/prestosql/plugin/hive/metastore/cache/TestCachingHiveMetastoreWithQueryRunner.java [hunk 1]
```diff
@@ -0,0 +1,119 @@
+package io.prestosql.plugin.hive.metastore.cache;
```
Comment (kokosing) [inline presto-hive-hadoop2/src/test/java/io/prestosql/plugin/hive/metastore/cache/TestCachingHiveMetastoreWithQueryRunner.java:1]: > CachingHiveMetastore didn't work correctly with GRANT/REVOKE

Instead of saying what was broken, please describe how it got improved. I would say:

```
Respect GRANT and REVOKE CachingHiveMetastore
```

File: presto-hive-hadoop2/src/test/java/io/prestosql/plugin/hive/metastore/cache/TestCachingHiveMetastoreWithQueryRunner.java [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+import java.util.Map;
+import java.util.Optional;
+
+import static com.google.common.io.RecursiveDeleteOption.ALLOW_INSECURE;
+import static io.prestosql.spi.security.SelectedRole.Type.ROLE;
+import static io.prestosql.testing.TestingSession.testSessionBuilder;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+@SuppressWarnings("UnstableApiUsage")   // for Closer
+public class TestCachingHiveMetastoreWithQueryRunner
```
Comment (kokosing) [inline presto-hive-hadoop2/src/test/java/io/prestosql/plugin/hive/metastore/cache/TestCachingHiveMetastoreWithQueryRunner.java:29]: extend `io.prestosql.testing.AbstractTestQueryFramework`

File: presto-hive-hadoop2/src/test/java/io/prestosql/plugin/hive/metastore/cache/TestCachingHiveMetastoreWithQueryRunner.java [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+                "hive.metastore", "file",
+                "hive.metastore.catalog.dir", tmp.toURI().toString(),
+                "hive.security", "sql-standard",
+                "hive.metastore-cache-ttl", "60m",
+                "hive.metastore-refresh-interval", "10m"));
+        queryRunner.execute(ADMIN, "CREATE SCHEMA " + SCHEMA);
+        queryRunner.execute("CREATE TABLE test (test INT)");
+    }
+
+    @AfterMethod
```
Comment (kokosing) [inline presto-hive-hadoop2/src/test/java/io/prestosql/plugin/hive/metastore/cache/TestCachingHiveMetastoreWithQueryRunner.java:62]: `AfterMethod(alwaysRun =true)`

File: presto-hive-hadoop2/src/test/java/io/prestosql/plugin/hive/metastore/cache/TestCachingHiveMetastoreWithQueryRunner.java [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+import static io.prestosql.testing.TestingSession.testSessionBuilder;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+@SuppressWarnings("UnstableApiUsage")   // for Closer
+public class TestCachingHiveMetastoreWithQueryRunner
+{
+    private static final String CATALOG = "test";
+    private static final String SCHEMA = "test";
+    private static final Session ADMIN = getTestSession(new Identity.Builder("admin")
+            .withRole(CATALOG, new SelectedRole(ROLE, Optional.of("admin"))).build());
```
Comment (kokosing) [inline presto-hive-hadoop2/src/test/java/io/prestosql/plugin/hive/metastore/cache/TestCachingHiveMetastoreWithQueryRunner.java:34]:
```suggestion
    private static final Session ADMIN = getTestSession(Identity.forUser("admin")
            .withRole(CATALOG, new SelectedRole(ROLE, Optional.of("admin")))
            .build());
```

File: presto-hive-hadoop2/src/test/java/io/prestosql/plugin/hive/metastore/cache/TestCachingHiveMetastoreWithQueryRunner.java [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+            throws Exception
+    {
+        queryRunner = DistributedQueryRunner
+                .builder(ADMIN)
+                .setNodeCount(1)
+                .build();
+        queryRunner.installPlugin(new HiveHadoop2Plugin());
+        File tmp = Files.createTempDir();
+        closer.register(() -> MoreFiles.deleteRecursively(tmp.toPath(), ALLOW_INSECURE));
+        queryRunner.createCatalog(CATALOG, "hive-hadoop2", Map.of(
```
Comment (kokosing) [inline presto-hive-hadoop2/src/test/java/io/prestosql/plugin/hive/metastore/cache/TestCachingHiveMetastoreWithQueryRunner.java:52]: ImmutableMap

File: presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/file/FileHiveMetastore.java [hunk 6]
```diff
@@ -1054,9 +1056,11 @@ public synchronized void grantTablePrivileges(String databaseName, String tableN
     public synchronized void revokeTablePrivileges(String databaseName, String tableName, String tableOwner, HivePrincipal grantee, Set<HivePrivilegeInfo> privileges)
     {
         Set<HivePrivilegeInfo> currentPrivileges = listTablePrivileges(databaseName, tableName, tableOwner, Optional.of(grantee));
-        currentPrivileges.removeAll(privileges);
+        Set<HivePrivilegeInfo> privilegesToRemove = privileges.stream()
+                .map(p -> new HivePrivilegeInfo(p.getHivePrivilege(), p.isGrantOption(), p.getGrantor(), grantee))
+                .collect(toUnmodifiableSet());
 
-        setTablePrivileges(grantee, databaseName, tableName, currentPrivileges);
+        setTablePrivileges(grantee, databaseName, tableName, Sets.difference(currentPrivileges, privilegesToRemove));
```
Comment (kokosing) [inline presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/file/FileHiveMetastore.java:1063]: Please fix file based metastore in separate commit

File: presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/cache/CachingHiveMetastore.java [hunk 7]
```diff
@@ -250,6 +250,7 @@ public void flushCache()
         tableStatisticsCache.invalidateAll();
         partitionStatisticsCache.invalidateAll();
         rolesCache.invalidateAll();
+        roleGrantsCache.invalidateAll();
```
Comment (kokosing) [inline presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/cache/CachingHiveMetastore.java:253]: This does not look like related to `GRANT/REVOKE`. Please extract this as separate commit.

File: presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/cache/CachingHiveMetastore.java [hunk 8]
```diff
@@ -888,6 +890,7 @@ public void revokeTablePrivileges(String databaseName, String tableName, String
         }
         finally {
             tablePrivilegesCache.invalidate(new UserTableKey(Optional.of(grantee), databaseName, tableName, tableOwner));
+            tablePrivilegesCache.invalidate(new UserTableKey(Optional.empty(), databaseName, tableName, tableOwner));
```
Comment (kokosing) [inline presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/cache/CachingHiveMetastore.java:893]: what about grant and revoke for roles?

File: presto-hive-hadoop2/src/test/java/io/prestosql/plugin/hive/metastore/cache/TestCachingHiveMetastoreWithQueryRunner.java [hunk 9]
```diff
@@ -0,0 +1,119 @@
...
+        queryRunner.execute(ALICE, "SELECT * FROM test");
+        queryRunner.execute("REVOKE SELECT ON test FROM " + ALICE_NAME);
+        assertThatThrownBy(() -> queryRunner.execute(ALICE, "SELECT * FROM test"))
+                .hasMessageContaining("Access Denied");
+    }
+
+    @DataProvider
+    private Object[][] testCacheRefreshOnRoleGrantAndRevokeParams()
+    {
+        String grantSelectStatement = "GRANT SELECT ON test TO ROLE test_role";
```
Comment (kokosing) [inline presto-hive-hadoop2/src/test/java/io/prestosql/plugin/hive/metastore/cache/TestCachingHiveMetastoreWithQueryRunner.java:93]: It is strange that you GRANT something that is created in test (line 112).

File: presto-hive-hadoop2/src/test/java/io/prestosql/plugin/hive/metastore/cache/TestCachingHiveMetastoreWithQueryRunner.java [hunk 10]
```diff
@@ -0,0 +1,119 @@
...
+        String grantRoleStatement = "GRANT test_role TO " + ALICE_NAME;
+        List<List<String>> grantRoleStatements = ImmutableList.of(
+                ImmutableList.of(grantSelectStatement, grantRoleStatement),
```
Comment (kokosing) [inline presto-hive-hadoop2/src/test/java/io/prestosql/plugin/hive/metastore/cache/TestCachingHiveMetastoreWithQueryRunner.java:96]: I would test GRANT table privileges and GRANT role independently. Now it is not known what caused cache to flush. We need to make sure that cache is flushed properly for any of these.

File: presto-hive-hadoop2/src/test/java/io/prestosql/plugin/hive/metastore/cache/TestCachingHiveMetastoreWithQueryRunner.java [hunk 11]
```diff
@@ -0,0 +1,119 @@
...
+        assertThatThrownBy(() -> queryRunner.execute(ALICE, "SELECT * FROM test"))
+                .hasMessageContaining("Access Denied");
+        queryRunner.execute("GRANT SELECT ON test TO " + ALICE_NAME);
+        queryRunner.execute(ALICE, "SELECT * FROM test");
+        queryRunner.execute("REVOKE SELECT ON test FROM " + ALICE_NAME);
+        assertThatThrownBy(() -> queryRunner.execute(ALICE, "SELECT * FROM test"))
+                .hasMessageContaining("Access Denied");
+    }
+
+    @DataProvider
```
Comment (kokosing) [inline presto-hive-hadoop2/src/test/java/io/prestosql/plugin/hive/metastore/cache/TestCachingHiveMetastoreWithQueryRunner.java:90]: Let's define data provider under the usage.

File: presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/file/FileHiveMetastore.java (hunk 6 above)
Comment (iirekm) [reply presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/file/FileHiveMetastore.java:1063]: will split into 3 commits a moment before merge, doing this on every fix is clumsy

File: presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/cache/CachingHiveMetastore.java (hunk 8 above)
Comment (iirekm) [reply presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/cache/CachingHiveMetastore.java:899]: works, see the test

File: presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/cache/CachingHiveMetastore.java (hunk 7 above)
Comment (iirekm) [reply presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/cache/CachingHiveMetastore.java]: will split into 3 commits a moment before merge, doing this on every fix is clumsy

File: presto-hive-hadoop2/src/test/java/io/prestosql/plugin/hive/metastore/cache/TestCachingHiveMetastoreWithQueryRunner.java (hunk 1 above)
Comment (iirekm) [reply presto-hive-hadoop2/src/test/java/io/prestosql/plugin/hive/metastore/cache/TestCachingHiveMetastoreWithQueryRunner.java:1]: will split into 3 commits a moment before merge, doing this on every fix is clumsy

File: presto-hive-hadoop2/src/test/java/io/prestosql/plugin/hive/metastore/cache/TestCachingHiveMetastoreWithQueryRunner.java (hunk 2 above)
Comment (iirekm) [reply presto-hive-hadoop2/src/test/java/io/prestosql/plugin/hive/metastore/cache/TestCachingHiveMetastoreWithQueryRunner.java:29]: I don't need any functionality from this base class

also as we discussed internally, reusing test code via inheritance does more harm than good

File: presto-hive-hadoop2/src/test/java/io/prestosql/plugin/hive/metastore/cache/TestCachingHiveMetastoreWithQueryRunner.java (hunk 3 above)
Comment (iirekm) [reply presto-hive-hadoop2/src/test/java/io/prestosql/plugin/hive/metastore/cache/TestCachingHiveMetastoreWithQueryRunner.java]: forgot

this is another reason why JUnit 5 is better, their `@After` method don't have such distinctions

File: presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/cache/CachingHiveMetastore.java (hunk 7 above)
Comment (lhofhansl) [reply presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/cache/CachingHiveMetastore.java:253]: And what about: grantedPrincipalsCache (my bad), and grantedPrincipalsCache?

File: presto-hive-hadoop2/src/test/java/io/prestosql/plugin/hive/metastore/cache/TestCachingHiveMetastoreWithQueryRunner.java (hunk 10 above)
Comment (iirekm) [reply presto-hive-hadoop2/src/test/java/io/prestosql/plugin/hive/metastore/cache/TestCachingHiveMetastoreWithQueryRunner.java:110]: why independently? those two steps are necessary for user to get privileges, they must be together; and for `GRANT SELEC ON ... to user` is separate test

File: presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/cache/CachingHiveMetastore.java (hunk 8 above)
Comment (lhofhansl) [reply presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/cache/CachingHiveMetastore.java:893]: This seems to be the crux of the change. It would be cool @iirekm, if that could be explained in the description - so we do not have guess from the code what the goal is :)

Comment (iirekm) [reply presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/cache/CachingHiveMetastore.java:899]: what you mean by "explained in description" - a comment?

the reason of this strange bahavior is that sometimes cache.get is called Optional.of(grantee), sometimes with Optional.empty(), for me this is a bigger issue for another task to simplify and refactory this code

File: presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/cache/CachingHiveMetastore.java [hunk 12]
```diff
@@ -218,38 +230,33 @@ public PartitionStatistics load(WithIdentity<HivePartitionName> key)
...
-        partitionNamesCache.invalidateAll();
-        databaseCache.invalidateAll();
-        tableCache.invalidateAll();
-        partitionCache.invalidateAll();
-        partitionFilterCache.invalidateAll();
-        tablePrivilegesCache.invalidateAll();
-        tableStatisticsCache.invalidateAll();
-        partitionStatisticsCache.invalidateAll();
-        rolesCache.invalidateAll();
+        caches.forEach(Cache::invalidateAll);
```
Comment (electrum) [inline presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/cache/CachingHiveMetastore.java:259]: I don’t see why we need to duplicate the list of caches just to invalidate them. The old code seems fine. This change is unrelated to fixing grant/revoke, so please make a separate PR and we can discuss it independently.

Comment (iirekm) [reply presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/cache/CachingHiveMetastore.java:259]: The old code was buggy, 4 caches weren't cleaned, so I added this to prevent such things in the future. Now the title of this pr will be different.

File: presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/cache/CachingHiveMetastore.java (hunk 8 above)
Comment (lhofhansl) [reply presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/cache/CachingHiveMetastore.java:899]: Thanks.

I meant to update the description of this PR with what you just said. But this comment explains it too.

Strange, though, what does a get without a specific grantee even mean?

File: presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/cache/CachingHiveMetastore.java (hunk 7 above)
Comment (iirekm) [reply presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/cache/CachingHiveMetastore.java]: in fact 4 caches weren't cleaned here :-(
I added cleaning in a loop to solve this once and for all

File: presto-hive-hadoop2/src/test/java/io/prestosql/plugin/hive/metastore/cache/TestCachingHiveMetastoreWithQueryRunner.java [hunk 13]
```diff
@@ -0,0 +1,134 @@
...
+    public void createQueryRunner()
+            throws Exception
+    {
+        queryRunner = DistributedQueryRunner
+                .builder(ADMIN)
+                .setNodeCount(1)
+                .build();
+        queryRunner.installPlugin(new HiveHadoop2Plugin());
+        File tmp = Files.createTempDir();
+        closer.register(() -> MoreFiles.deleteRecursively(tmp.toPath(), ALLOW_INSECURE));
```
Comment (kokosing) [inline presto-hive-hadoop2/src/test/java/io/prestosql/plugin/hive/metastore/cache/TestCachingHiveMetastoreWithQueryRunner.java:66]: static import,
closer for single usage is an overkill to me.

File: presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/cache/CachingHiveMetastore.java (hunk 12 above)
Comment (kokosing) [reply presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/cache/CachingHiveMetastore.java:259]: now it is buggy in different way. Still someone might forget to add a cache to `caches`. Please undo the refactor or `please make a separate PR and we can discuss it`

File: presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/cache/CachingHiveMetastore.java (hunk 8 above)
Comment (kokosing) [reply presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/cache/CachingHiveMetastore.java:899]: `listTablePrivileges` is called with `Optional.empty()` for `principal` when we want to access all privileges. It is set when when we want to access specific user privileges.

I think adding a comment to code could be nice here.

File: presto-hive-hadoop2/src/test/java/io/prestosql/plugin/hive/metastore/cache/TestCachingHiveMetastoreWithQueryRunner.java [hunk 14]
```diff
@@ -0,0 +1,134 @@
+/*
```
Comment (kokosing) [inline presto-hive-hadoop2/src/test/java/io/prestosql/plugin/hive/metastore/cache/TestCachingHiveMetastoreWithQueryRunner.java:1]: Commit message:
`Add missing cache invalidate calls to CachingHiveMetastore` is implementation detail, however the more important aspect is user visiable behavior change. Also please do not use `-` in commit message title.

I would write:
```
Invalidate Hive metastore cache on GRANT/REVOKE

- Add missing cache invalidate calls to CachingHiveMetastore
- Fix REVOKE in FileHiveMetastore
```

File: presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/file/FileHiveMetastore.java (hunk 6 above)
Comment (kokosing) [reply presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/file/FileHiveMetastore.java:1062]: > will split into 3 commits a moment before merge

A person who merge and so accepts the code has to see the code in a form it is going to be merged.

Anyway, I think it could stay as it is now. Single commit is not perfect here, but it is good enough.

File: presto-hive-hadoop2/src/test/java/io/prestosql/plugin/hive/metastore/cache/TestCachingHiveMetastoreWithQueryRunner.java (hunk 2 above)
Comment (kokosing) [reply presto-hive-hadoop2/src/test/java/io/prestosql/plugin/hive/metastore/cache/TestCachingHiveMetastoreWithQueryRunner.java:43]: Just FYI there is io.prestosql.sql.query.QueryAssertions

File: presto-hive-hadoop2/src/test/java/io/prestosql/plugin/hive/metastore/cache/TestCachingHiveMetastoreWithQueryRunner.java [hunk 15]
```diff
@@ -0,0 +1,134 @@
...
+    private static final Session ALICE = getTestSession(new Identity.Builder(ALICE_NAME).build());
+
+    private DistributedQueryRunner queryRunner;
+    private final Closer closer = Closer.create();
+
+    @BeforeMethod
+    public void createQueryRunner()
+            throws Exception
+    {
+        queryRunner = DistributedQueryRunner
```
Comment (kokosing) [inline presto-hive-hadoop2/src/test/java/io/prestosql/plugin/hive/metastore/cache/TestCachingHiveMetastoreWithQueryRunner.java:60]: Use `io.prestosql.plugin.hive.HiveQueryRunner#builder` then you don't have to play with metastore directory.

File: presto-hive-hadoop2/src/test/java/io/prestosql/plugin/hive/metastore/cache/TestCachingHiveMetastoreWithQueryRunner.java [hunk 16]
```diff
@@ -0,0 +1,134 @@
...
+                "hive.metastore-refresh-interval", "10m"));
+        queryRunner.execute(ADMIN, "CREATE SCHEMA " + SCHEMA);
+        queryRunner.execute("CREATE TABLE test (test INT)");
+    }
+
+    @AfterMethod(alwaysRun = true)
+    void cleanUp()
+            throws IOException
+    {
+        closer.close();
```
Comment (kokosing) [inline presto-hive-hadoop2/src/test/java/io/prestosql/plugin/hive/metastore/cache/TestCachingHiveMetastoreWithQueryRunner.java:81]: you need to close query runner here.

File: presto-hive-hadoop2/src/test/java/io/prestosql/plugin/hive/metastore/cache/TestCachingHiveMetastoreWithQueryRunner.java [hunk 17]
```diff
@@ -0,0 +1,134 @@
...
+    }
+
+    static Session getTestSession(Identity identity)
```
Comment (kokosing) [inline presto-hive-hadoop2/src/test/java/io/prestosql/plugin/hive/metastore/cache/TestCachingHiveMetastoreWithQueryRunner.java:84]: private static

File: presto-hive-hadoop2/src/test/java/io/prestosql/plugin/hive/metastore/cache/TestCachingHiveMetastoreWithQueryRunner.java [hunk 18]
```diff
@@ -0,0 +1,134 @@
...
+                "hive.metastore.catalog.dir", tmp.toURI().toString(),
+                "hive.security", "sql-standard",
+                "hive.metastore-cache-ttl", "60m",
+                "hive.metastore-refresh-interval", "10m"));
+        queryRunner.execute(ADMIN, "CREATE SCHEMA " + SCHEMA);
+        queryRunner.execute("CREATE TABLE test (test INT)");
+    }
+
+    @AfterMethod(alwaysRun = true)
+    void cleanUp()
```
Comment (kokosing) [inline presto-hive-hadoop2/src/test/java/io/prestosql/plugin/hive/metastore/cache/TestCachingHiveMetastoreWithQueryRunner.java:78]: public

File: presto-hive-hadoop2/src/test/java/io/prestosql/plugin/hive/metastore/cache/TestCachingHiveMetastoreWithQueryRunner.java [hunk 19]
```diff
@@ -0,0 +1,134 @@
...
+        String grantRoleStatement = "GRANT test_role TO " + ALICE_NAME;
+        List<List<String>> grantRoleStatements = ImmutableList.of(
+                ImmutableList.of(grantSelectStatement, grantRoleStatement),
+                ImmutableList.of(grantRoleStatement, grantSelectStatement));
+        List<String> revokeRoleStatements = ImmutableList.of(
+                "DROP ROLE test_role",
+                "REVOKE SELECT ON test FROM ROLE test_role",
+                "REVOKE test_role FROM " + ALICE_NAME);
+        return grantRoleStatements.stream()
+                .flatMap(arg1 -> revokeRoleStatements.stream().map(arg2 -> new Object[] {arg1, arg2}))
```
Comment (kokosing) [inline presto-hive-hadoop2/src/test/java/io/prestosql/plugin/hive/metastore/cache/TestCachingHiveMetastoreWithQueryRunner.java:131]: `com.google.common.collect.Sets#cartesianProduct(java.util.Set<? extends B>...)`?

File: presto-hive-hadoop2/src/test/java/io/prestosql/plugin/hive/metastore/cache/TestCachingHiveMetastoreWithQueryRunner.java (hunk 2 above)
Comment (iirekm) [reply presto-hive-hadoop2/src/test/java/io/prestosql/plugin/hive/metastore/cache/TestCachingHiveMetastoreWithQueryRunner.java:43]: I just need to know if query passes or fails in this test, so no assertion utility class is needed for me, just regular method calls

File: presto-hive-hadoop2/src/test/java/io/prestosql/plugin/hive/metastore/cache/TestCachingHiveMetastoreWithQueryRunner.java (hunk 14 above)
Comment (iirekm) [reply presto-hive-hadoop2/src/test/java/io/prestosql/plugin/hive/metastore/cache/TestCachingHiveMetastoreWithQueryRunner.java:1]: splitted into 2 commits

File: presto-hive-hadoop2/src/test/java/io/prestosql/plugin/hive/metastore/cache/TestCachingHiveMetastoreWithQueryRunner.java (hunk 1 above)
Comment (iirekm) [reply presto-hive-hadoop2/src/test/java/io/prestosql/plugin/hive/metastore/cache/TestCachingHiveMetastoreWithQueryRunner.java:14]: splitted into 2 commits

File: presto-hive-hadoop2/src/test/java/io/prestosql/plugin/hive/metastore/cache/TestCachingHiveMetastoreWithQueryRunner.java (hunk 15 above)
Comment (iirekm) [reply presto-hive-hadoop2/src/test/java/io/prestosql/plugin/hive/metastore/cache/TestCachingHiveMetastoreWithQueryRunner.java:60]: my setup is completely different from the offering of this builder, code reuse in this case makes no sense

File: presto-hive-hadoop2/src/test/java/io/prestosql/plugin/hive/metastore/cache/TestCachingHiveMetastoreWithQueryRunner.java (hunk 13 above)
Comment (iirekm) [reply presto-hive-hadoop2/src/test/java/io/prestosql/plugin/hive/metastore/cache/TestCachingHiveMetastoreWithQueryRunner.java]: done

File: presto-hive-hadoop2/src/test/java/io/prestosql/plugin/hive/metastore/cache/TestCachingHiveMetastoreWithQueryRunner.java (hunk 18 above)
Comment (iirekm) [reply presto-hive-hadoop2/src/test/java/io/prestosql/plugin/hive/metastore/cache/TestCachingHiveMetastoreWithQueryRunner.java]: added for convention, it matters on JUnit, but it doesn't matter in testng

File: presto-hive-hadoop2/src/test/java/io/prestosql/plugin/hive/metastore/cache/TestCachingHiveMetastoreWithQueryRunner.java (hunk 16 above)
Comment (iirekm) [reply presto-hive-hadoop2/src/test/java/io/prestosql/plugin/hive/metastore/cache/TestCachingHiveMetastoreWithQueryRunner.java]: done

File: presto-hive-hadoop2/src/test/java/io/prestosql/plugin/hive/metastore/cache/TestCachingHiveMetastoreWithQueryRunner.java (hunk 17 above)
Comment (iirekm) [reply presto-hive-hadoop2/src/test/java/io/prestosql/plugin/hive/metastore/cache/TestCachingHiveMetastoreWithQueryRunner.java]: done

File: presto-hive-hadoop2/src/test/java/io/prestosql/plugin/hive/metastore/cache/TestCachingHiveMetastoreWithQueryRunner.java (hunk 9 above)
Comment (iirekm) [reply presto-hive-hadoop2/src/test/java/io/prestosql/plugin/hive/metastore/cache/TestCachingHiveMetastoreWithQueryRunner.java:121]: test table is created in `@BeforeMethod`, so it's quite reasonable

also I wouldn't be surprised if an implementation of "GRANT" would rely not on particular object but on object name, in that case it could even work before creating table

File: presto-hive-hadoop2/src/test/java/io/prestosql/plugin/hive/metastore/cache/TestCachingHiveMetastoreWithQueryRunner.java (hunk 19 above)
Comment (iirekm) [reply presto-hive-hadoop2/src/test/java/io/prestosql/plugin/hive/metastore/cache/TestCachingHiveMetastoreWithQueryRunner.java]: done

File: presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/cache/CachingHiveMetastore.java (hunk 8 above)
Comment (iirekm) [reply presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/cache/CachingHiveMetastore.java]: done

File: presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/file/FileHiveMetastore.java (hunk 6 above)
Comment (iirekm) [reply presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/file/FileHiveMetastore.java:1062]: done

File: presto-hive-hadoop2/src/test/java/io/prestosql/plugin/hive/metastore/cache/TestCachingHiveMetastoreWithQueryRunner.java [hunk 20]
```diff
@@ -0,0 +1,133 @@
...
+
+import static com.google.common.collect.Lists.cartesianProduct;
+import static com.google.common.io.Files.createTempDir;
+import static com.google.common.io.MoreFiles.deleteRecursively;
+import static com.google.common.io.RecursiveDeleteOption.ALLOW_INSECURE;
+import static io.prestosql.spi.security.SelectedRole.Type.ROLE;
+import static io.prestosql.testing.TestingSession.testSessionBuilder;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+@SuppressWarnings("UnstableApiUsage")   // for Closer
```
Comment (kokosing) [inline presto-hive-hadoop2/src/test/java/io/prestosql/plugin/hive/metastore/cache/TestCachingHiveMetastoreWithQueryRunner.java:41]: This is no longer needed

**Comment by kokosing** [comment]
Thanks!

File: presto-hive-hadoop2/src/test/java/io/prestosql/plugin/hive/metastore/cache/TestCachingHiveMetastoreWithQueryRunner.java (hunk 20 above)
Comment (iirekm) [reply presto-hive-hadoop2/src/test/java/io/prestosql/plugin/hive/metastore/cache/TestCachingHiveMetastoreWithQueryRunner.java:41]: needed - other `@Beta` stuff from Guava used

**Comment by kokosing** [comment]
Merged, thanks!
